overlord/devicestate/devicestate.go

```diff
@@ -556,7 +590,19 @@ func prereqsFromSnapTaskSet(ts *state.TaskSet) ([]string, error) {
 	return nil, fmt.Errorf("internal error: cannot identify task-snap-setup in taskset")
 }
 
-func remodelTasks(ctx context.Context, st *state.State, current, new *asserts.Model, deviceCtx snapstate.DeviceContext, fromChange string) ([]*state.TaskSet, error) {
+func sideInfoAndPathFromID(sis []*snap.SideInfo, paths []string, id string) (*snap.SideInfo, string) {
+	for i, si := range sis {
+		if si.SnapID == id {
```

Comment:
this probably needs a comment because this will not be set for unsasserted snaps, I suppose we are not supporting that for now?